core/src/mos6502.cpp

```diff
@@ -261,6 +261,23 @@ Pipeline Mos6502::parse_next_instruction() {
     case Instruction::NOP:
         result.push([]() { /* Do nothing. */ });
         break;
+    case Instruction::INC: {
+        if (opcode.address_mode == AddressMode::Zeropage) {
+            result.append(create_zeropage_addressing_steps());
+            result.push([=]() { tmp_ = mmu_->read_byte(effective_address_); });
+            result.push(
+                    [=]() { mmu_->write_byte(effective_address_, tmp_++); });
```

Comment:
I think some of the instructions we have implemented already skips similar extra writes. We should probably go through them again later and fix that :)

Reply:
You're right. :) Added an [issue](#107) for it.